pywemo/util.py

```diff
@@ -54,3 +55,15 @@ def interface_addresses():
             addresses.append(addr.ip)
 
     return addresses
+
+
+def get_ip_address(host='1.2.3.4'):
+    """Return IP from hostname or IP."""
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        sock.connect((host, 9))
+        return sock.getsockname()[0]
+    except socket.error:
+        return None
+    finally:
+        del sock
```

Comment:
I know this function was just moved and isn't new, but should this call `sock.close` before it is deleted?  Or are there cases would calling close would raise another error here?

Reply:
Both `del sock` and `sock.close` end up closing the socket.

**del**: https://github.com/python/cpython/blob/0b858cdd5d114f0890b11b6c4d6559d0ceb468ab/Modules/socketmodule.c#L4993

**close**: https://github.com/python/cpython/blob/0b858cdd5d114f0890b11b6c4d6559d0ceb468ab/Modules/socketmodule.c#L3136

 I think `sock.close` is more explicit about what is intended. Would you prefer one be used over the other going forward?

Reply:
I prefer the `close` method myself, I just wanted to make sure there wasn't anything missed by not actually deleting it since I saw a few examples online as well where they deleted it (per my other comment further below)